rdr_service/tools/tool_libs/consent_validation_report.py

```diff
@@ -0,0 +1,508 @@
+                           'Signature Date Mismatch',
+                           'Invalid DOB',
+                           'Age at Primary Consent <18',
+                           'Checkbox Unchecked',
+                           'Non-VA Consent for VA Participant',
+                           'VA Consent for Non-VA Participant'],
+                'format': {'textFormat': {'bold': True},
+                           'wrapStrategy': 'WRAP',
+                           'borders': {'top': {'style': 'SOLID'},'bottom': {'style': 'SOLID'}},
+                           'backgroundColor': {"red": 0.02, "green": 0.8, "blue": 0.4},
+                           'verticalAlignment': 'MIDDLE'
+                           }
+
+            },
+        }
+
+    def get_daily_consent_validation_results(self, db_conn=None):
+        """
+        Queries the RDR consent_file table and populates the pandas DataFrame with the validation results
+        from the specified report date (records with matching created date).   Sets self.daily_data to the dataframe
+
+        Notes on the DAILY_REPORT_SQL logic, to prevent flagging errors that should only be counted when there wasn't
+        a higher-level error related to the consent:
+        - A signature_missing error will only be true when the consent file exists
+        - An invalid_signing_date error will only be true when the signature is valid (exists)
+        """
+        if not db_conn:
+            raise(EnvironmentError, 'No active DB connection object')
+
+        DAILY_REPORT_SQL = """
+            SELECT cf.participant_id,
+                   p.date_of_birth,
+                   -- age_at_consent is specific/relevant to PRIMARY consent type (uses its authored date)
+                   CASE WHEN cf.type = 1 THEN
+                        CASE WHEN (p.date_of_birth IS NULL or p.consent_for_study_enrollment_authored IS NULL) THEN NULL
+                             ELSE TIMESTAMPDIFF(YEAR, p.date_of_birth, p.consent_for_study_enrollment_authored)
+                        END
+                   END AS age_at_consent,
+                   CASE
+                      WHEN (h.name IS NOT NULL and h.name != 'UNSET') THEN h.name
+                      ELSE '(No HPO details)'
+                   END AS hpo,
+                   CASE
+                      WHEN o.display_name IS NOT NULL THEN o.display_name ELSE '(No organization details)'
+                   END AS organization,
+                   cf.sync_status,
+                   cf.type,
+                   cf.file_path,
+                   -- CASE Statements to calculate the known tracked error conditions
+                   CASE WHEN cf.file_exists = 0 THEN 1 ELSE 0 END AS missing_file,
+                   CASE WHEN (cf.file_exists and cf.is_signature_valid = 0) THEN 1 ELSE 0 END AS signature_missing,
+                   CASE WHEN (cf.is_signature_valid and cf.is_signing_date_valid = 0) THEN 1 ELSE 0
+                   END AS invalid_signing_date,
```

Comment:
You said that is_signature_valid means a signature exists;  without a signature, should we even be trying to validate the date?  My concern is overcounting the invalid date errors because there is no data  at all for the signature line, vs. a bad date associated with a valid signature

Reply:
True for `is_signature_valid` does mean a signature exists, but the signature text/image and date string are independent from each other in the PDF structure and it's possible to receive the date without a signature. I'm not sure of the best way to keep the numbers mutually exclusive.

Reply:
The `signing_date` field is extracted from the PDF?, So, should I change it to:
`(cf.signing_date IS NOT NULL and NOT cf.is_signing_date_valid) AS invalid_signing_date`

Do we need separate error tracking for 
`(cf.is_signature_valid and cf.signing_date IS NULL) AS 'undated_signature'` or something?

Reply:
I don't know. It might make more sense to just ignore the invalid date counts if they're being counted in the signature errors. I just wanted to point it out in case it was something you hadn't thought of yet.

Reply:
If you're saying that the two fields (signature and signing date) are extracted independently from the PDF then I think it makes sense to count their errors independently.   But I think calculation of an `invalid_signing_date` is predicated on there being a s`signing_date ` value.  Right now there are ~4800 records without a` signing_date` .  I will ask Megan if we need another error type for missing dates. 

Are you ready to approve the proposed logic change above?

Reply:
Based on offline conversation, going with the original logic:  One error bucket ('invalid_signature`) for missing signatures in PDF and one for any invalid signing dates (including missing a date) if a signature is present.  I will push up a change to rename the report column for clarity (from "Signature Date Mismatch" to "Signature Date Invalid")